### DISCUSSION TITLE
Capturing ISS trail

### DISCUSSION THREAD

**Discussion by evripidis** [aaronwmorris/indi-allsky#1156] [Ideas]
Just a crazy idea...

I was just thinking....we do know ISS trail in real time and we can predict it (for example https://transit-finder.com)

If we had a way to use those data to capture it automatically, changing the capture settings.

Dummy " If the ISS is less than 50km from my location, then change the capture settings, or record video etc,"

**Comment by aaronwmorris** [comment]
I believe you upgraded from an older release, but the latest releases of indi-allsky do track major satellites.  There is image label/overlay data for ISS and Hubble, and today I just added the Tiangong Chinese space station.

You can update your overlay using the default from the wiki to see this new data:

https://github.com/aaronwmorris/indi-allsky/wiki/Image-Labels

You will have to pull down the latest code to get the Tiangong data.

**Reply by leforet** [reply]
Hi, is there a way to dynamically change the list based on the time frame of what is passing overhead?   So when an image is being captured for XX time, it does a lookup for the predicted passes, above a certain magnitude and applies the satellite name to the orbiting objects.
https://www.satflare.com/track.asp?q=visual#LIST

![SatelliteOverHead](https://github.com/user-attachments/assets/91cec157-6db3-4dbb-ad71-d15f2484ebbb)

**Reply by aaronwmorris** [reply]
Within indi-allsky, there is data for over 6600 satellites currently.  Space stations, 100 brightest, and all starlink are currently populated.  The challenge is it requires a finite amount of CPU time to calculate the position of each satellite.  For a slower system, such as Raspberry Pi 3, using pyephem, it requires about 5-8ms for each satellite.  (Skyfield is slower at 50-60ms).

At 5ms, it requires a full 30s to calculate the position of 6600 satellites.  Calculating the position only gives the position at that moment in time... the position has to be recalculated for each image.  It is much more manageable if only the 100 brightest are calculated.

"100 brightest" refers to this list at Celestrak:  https://celestrak.org/NORAD/elements/table.php?GROUP=visual&FORMAT=tle

I will investigate adding a list of visible satellites from the 100 brightest list to the image overlay.

**Reply by leforet** [reply]
thanks,  
the reason I ask is that I had been getting a satellite flare (right hand side of image, halfway up, near the edge the sky near the trees) from what I think are StarLink satellites.  Only seems to occur in the early evening.   The two images from all sky have a flare  roughly in the same orbital path
![ccd1_20240927_205159 (1)](https://github.com/user-attachments/assets/ead7c8df-88e6-405d-a63a-468f7eaaa54b)
![ccd1_20240927_204615 (1)](https://github.com/user-attachments/assets/7907cf60-d09f-46b2-9812-6188959dd6fe)

overhead Starlink orbits which I think is the culprit 
![StarLinkOrbitPath4](https://github.com/user-attachments/assets/fd3ecd1f-a4a7-4aeb-a7bd-7c752c32df62)

**Comment by evripidis** [comment]
Yeap ok, I can see that now.

So I just leave the second part of my idea, if there would a way to auto change capture settings when ISS is passing by :-)

**Reply by aaronwmorris** [reply]
What kind of settings change would you be looking for?  Capturing a video would not be possible, though.

**Reply by caphector** [reply]
I'd been wondering if any satellites were bright enough to be visible if the shutter was open when flying overhead. Could you ensure the shutter is open during transit to see if it shows up? Not sure if it's bright enough for these cameras.

**Reply by aaronwmorris** [reply]
I have given this a bit of thought... so here is the challenge with this idea.

The exposure the camera takes is based on the brightness of the sky/clouds/etc.  If sky conditions were perfect, you can extend the exposure to a point, but an ISS flyover can be anywhere from 1 to 5 minutes.  If your exposure is 60 seconds, trying to take a 300 second exposure, there is a pretty high risk of significantly over-exposing the image.  If conditions are less than perfect, the result will likely be unusable.

What would need to happen is dropping the gain before taking such a long exposure.  Dropping the gain has other requirements, such as needing proper dark frames [at that gain level] to remove all of the hot pixel noise.

I am not saying this is a bad idea, but it is probably not a great "first class" feature to incorporate into indi-allsky directly.  A dedicated capture program would probably be better.

**Comment by mburge** [comment]
Speaking of the ISS trail...

![image](https://github.com/aaronwmorris/indi-allsky/assets/3086001/4cc9cba4-e42a-4395-bb10-4639b16cbafb)


![image](https://github.com/aaronwmorris/indi-allsky/assets/3086001/100b5431-8427-4f51-b66b-f99d3ca466e5)
